advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,665 @@
+                loss = criterion(output, target)  
+                cnt += 1  
+                acc1, acc5 = accuracy(output, target, topk=(1, 5))  
+                print('.', end = '')  
+                top1.update(acc1[0], image.size(0)) 
+                top5.update(acc5[0], image.size(0)) 
+                if cnt >= neval_batches:  
+                     return top1, top5  
+
+        return top1, top5 
+
+    def load_model(model_file): 
+        model = MobileNetV2() 
+        state_dict = torch.load(model_file) 
+        model.load_state_dict(state_dict) 
+        model.to('cpu') 
+        return model  
+
+    def print_size_of_model(model): 
+        torch.save(model.state_dict(), "temp.p")  
+        print('Size (MB):', os.path.getsize("temp.p")/1e6)  
+        os.remove('temp.p') 
+
+3. Define dataset and data loaders  
+----------------------------------  
+ 
+As our last major setup step, we define our dataloaders for our training and testing set. 
+ 
+ImageNet Data 
+^^^^^^^^^^^^^ 
+
+Please download imagenet dataset by following the instructions at: TBD. The rest of the tutorial assumes that you have imagenet_1k.zip which contains the sample imagenet dataset.
+
+Unzip imagenet_1k.zip to `data` folder.
+
+For this tutorial to run, we download this data and move it to the right place using  
+`these lines <https://github.com/pytorch/tutorials/blob/master/Makefile#L97-L98>`_  
+from the `Makefile <https://github.com/pytorch/tutorials/blob/master/Makefile>`_. 
+
+To run the code in this tutorial using the entire ImageNet dataset, on the other hand, you could download 
+the data using ``torchvision`` following  
+`here <https://pytorch.org/docs/stable/torchvision/datasets.html#imagenet>`_. For example,  
+to download the training set and apply some standard transformations to it, you could use:  
+
+.. code:: python  
+
+    import torchvision  
+    import torchvision.transforms as transforms 
+
+    imagenet_dataset = torchvision.datasets.ImageNet( 
+        '~/.data/imagenet', 
+        split='train',  
+        download=True,  
+        transforms.Compose([  
+            transforms.RandomResizedCrop(224),  
+            transforms.RandomHorizontalFlip(),  
+            transforms.ToTensor(),  
+            transforms.Normalize(mean=[0.485, 0.456, 0.406],  
+                                 std=[0.229, 0.224, 0.225]),  
+        ])  
+
+With the data downloaded, we show functions below that define dataloaders we'll use to read 
+in this data. These functions mostly come from  
+`here <https://github.com/pytorch/vision/blob/master/references/detection/train.py>`_.
+
+
+.. code:: python
+
+    def prepare_data_loaders(data_path):  
+
+        traindir = os.path.join(data_path, 'train') 
+        valdir = os.path.join(data_path, 'val') 
+        normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],  
+                                         std=[0.229, 0.224, 0.225]) 
+
+        dataset = torchvision.datasets.ImageFolder( 
```

Comment:
@raghuramank100 we can only read from folder after the data is downloaded

Reply:
I think we can do this:
'First ask user to download the imagenet archive directly from imagenet, by following the instructions at "http://www.image-net.org/download". Once that is done, we define dataloaders as follows:
```
     def prepare_data_loaders(data_path):  
       normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],  
                                         std=[0.229, 0.224, 0.225])
 dataset = ImageNet(
        data_path, split="train",
      transforms.Compose([  
                transforms.RandomResizedCrop(224),  
                transforms.RandomHorizontalFlip(),  
                transforms.ToTensor(),  
                normalize,  
            ]))  
    dataset_test = ImageNet(
        root_path, split="val", 
            transforms.Compose([  
                transforms.Resize(256), 
                transforms.CenterCrop(224), 
                transforms.ToTensor(),  
                normalize,  
            ])) 
```

Reply:
I think we need to use torchvision.datasets.ImageNet because this does some additional work.